xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -1154,12 +1154,24 @@ void CActiveAESink::GenerateNoise()
 
 void CActiveAESink::SetSilenceTimer()
 {
-  if (m_extStreaming)
+  if (m_extStreaming) // handles playback
+  {
     m_extSilenceTimeout = std::chrono::milliseconds::max();
-  else if (m_extAppFocused)
-    m_extSilenceTimeout = m_silenceTimeOut;
+  }
+  else if (m_extAppFocused) // handles no playback/GUI and playback in pause and seek
+  {
+    // only true with AudioTrack RAW + passthrough + TrueHD or EAC3 (DD+)
+    const bool noSilenceOnPause =
+        !m_needIecPack && m_requestedFormat.m_dataFormat == AE_FMT_RAW &&
+        (m_sinkFormat.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_TRUEHD ||
+         m_sinkFormat.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_EAC3);
+
+    m_extSilenceTimeout = (noSilenceOnPause) ? 0ms : m_silenceTimeOut;
```

Comment:
Technically I still don't understand why this is only needed for TrueHD / DDPlus and not for e.g. AC3 - leave alone DTS / DTS-HD. Maybe the adb logcat reveals that this "forces" the sink to pause itself. If that's the case, then the code inside the AudioTrack sink to handle the AddPause calls is just wrong :-).

Do you have an idea why this works and only for TrueHD / DD+?

Reply:
I think TrueHD and DD+ would require some more implementation or change in AudioTrack RAW specific code. As commented previously this PR is only a workaround to fix one issue/bug ONLY present in TrueHD / DD+ RAW.

Obviously we can remove the conditions and do the same for all codecs in RAW, but then pause/seek that already works well in the rest of the codecs gets worse.

Probably the root cause is still related to fact in TrueHD  not all audio frames are playable by itself (requires previous major sync frames similar to video codecs requires key frame). On the contrary with DTS you can cut anywhere and play.

Reply:
> A Dolby TrueHD bit stream consists of a sequence of access units. An access unit consists of an
> MLP Sync (from label mlp_sync to label start in the syntax below) followed by a segment of each
> substream. A major sync is the same as a minor sync, but with additional information contained in
> the substructure major_sync_info(). Major syncs can be distinguished by the fact that the bitfield of
> 32 bits starting at bit offset 32 from the start of the access unit always equals 0xF8726FBA, whereas
> the corresponding bitfield of a minor sync can never take this value.

https://developer.dolby.com/globalassets/technology/dolby-truehd/dolbytruehdhighlevelbitstreamdescription.pdf

Reply:
**AddPause** is only used/implemented for Audiotrack sink, not used in Windows / Linux, etc:

![AddPause](https://user-images.githubusercontent.com/58434170/210134904-7d591cf5-5283-4d7f-ab4e-dbd11308ae9e.png)

It's probably a flawed design from the start and a "PackPauseRAW" should be implemented just like in IEC:

https://github.com/xbmc/xbmc/blob/939c521a00f72457ce441f3a90910e61962dc213/xbmc/cores/AudioEngine/Utils/AEBitstreamPacker.cpp#L111-L138

Coincidentally or not, there we have a different code path TrueHD and DD+

Reply:
> Do you have an idea why this works and only for TrueHD / DD+?

> For the AVR it is as if the stream has ended (while it's paused or seeking) and when unpaused it starts synchronizing with the next major_sync_frame.

Reply:
As said: AddPause is only there to indicate the time of a gap to introduce and fill. Which is done by putting the sink itself to pause(), meaning the android object of the audio track. Also android is the only sink which wants raw pt ... all others use the iec path.

And yes, we should not rely on a non documented behaviour.

Different question: does it also work if you just return from AddPause() in case for Truehd and DD+ after the usleep?

Reply:
Test it and answer yourself. I can't test it right now.

Reply:
Same here, sadly ... maybe I will buy a new laptop soon, so that I can compile kodi in under 3 hours again :-)